File: SDKs/beezup.com/2.0/typescript/src/sdk/models/operations/getchannelcatalogproductinfolist.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetChannelCatalogProductInfoListPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=channelCatalogId" })
  channelCatalogId: string;
}


export class GetChannelCatalogProductInfoListRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetChannelCatalogProductInfoListPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.GetChannelCatalogProductInfoListRequest;
}


export class GetChannelCatalogProductInfoListResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  beezUPCommonErrorResponseMessage?: shared.BeezUpCommonErrorResponseMessage;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  channelCatalogProductInfoList?: shared.ChannelCatalogProductInfoList;
}

File: SDKs/beezup.com/2.0/python/src/sdk/models/operations/getchannelcatalogproductinfolist.py
import dataclasses
from typing import Optional
from ..shared import getchannelcatalogproductinfolistrequest as shared_getchannelcatalogproductinfolistrequest
from ..shared import beezup_common_errorresponsemessage as shared_beezup_common_errorresponsemessage
from ..shared import channelcatalogproductinfolist as shared_channelcatalogproductinfolist


@dataclasses.dataclass
class GetChannelCatalogProductInfoListPathParams:
    channel_catalog_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'channelCatalogId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetChannelCatalogProductInfoListRequest:
    path_params: GetChannelCatalogProductInfoListPathParams = dataclasses.field()
    request: shared_getchannelcatalogproductinfolistrequest.GetChannelCatalogProductInfoListRequest = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class GetChannelCatalogProductInfoListResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    beez_up_common_error_response_message: Optional[shared_beezup_common_errorresponsemessage.BeezUpCommonErrorResponseMessage] = dataclasses.field(default=None)
    channel_catalog_product_info_list: Optional[shared_channelcatalogproductinfolist.ChannelCatalogProductInfoList] = dataclasses.field(default=None)
    
